AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -174,26 +174,30 @@ public final class APIClient: APIClientProtocol {
         _ request: R
     ) throws -> HTTPResponse<R.ResponseType> {
         log(result: result, request: request)
+        
         do {
             if result.data.isEmpty, let emptyResponse = EmptyResponse() as? R.ResponseType {
                 return HTTPResponse(
                     headers: result.headers,
                     statusCode: result.statusCode,
-                    responseBody: emptyResponse
+                    responseBody: emptyResponse,
+                    responseData: Data()
```

Comment:
maybe we can pass `result.data` here for `responseData`

Reply:
> Is it possible to add a test that verifies that the data in `responseData` is the same returned from the api?

Not very easily... ApiClient is not setup in such a way as to allow mocking the URLSession's http response that easily, For now i've added a test that somewhat verifies the responseData: I marshal the responseData into a given type and then compare the values for it against the responseBody object in the HTTPResponse, so it kinda validates that the info is the same, though not that it's **exactly** the same original data :/

Reply:
I've added a test to compare the responseBody from using the apiClient to the raw data response from calling the API directly with `urlSession.data(for: request)`